File: backend/src/main/java/com/devmatch/backend/global/app/AppConfig.java
package com.devmatch.backend.global.app;

import com.devmatch.backend.standard.util.Ut;
import com.fasterxml.jackson.databind.ObjectMapper;
import jakarta.annotation.PostConstruct;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.security.crypto.password.PasswordEncoder;

@Configuration
public class AppConfig {

  @Bean
  PasswordEncoder passwordEncoder() {
    return new BCryptPasswordEncoder();
  }

  private static ObjectMapper objectMapper;

  @Autowired
  public void setObjectMapper(ObjectMapper objectMapper) {
    AppConfig.objectMapper = objectMapper;
  }

  @PostConstruct
  public void postConstruct() {
    Ut.json.objectMapper = objectMapper;
  }
}

File: backend/src/main/java/com/devmatch/backend/global/app/AppConfig.kt
package com.devmatch.backend.global.app

import com.devmatch.backend.standard.util.Ut
import com.fasterxml.jackson.databind.ObjectMapper
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.security.crypto.password.PasswordEncoder

@Configuration
class AppConfig(
    objectMapper: ObjectMapper
) {
    init{
        Ut.json.objectMapper = objectMapper
    }

    @Bean
    fun passwordEncoder(): PasswordEncoder {
        return BCryptPasswordEncoder()
    }
}
